pkg/handlers/generic/mutation/extraapiservercertsans/inject.go

```diff
@@ -50,12 +61,16 @@ func (h *extraAPIServerCertSANsPatchHandler) Mutate(
 	obj *unstructured.Unstructured,
 	vars map[string]apiextensionsv1.JSON,
 	holderRef runtimehooksv1.HolderReference,
-	_ client.ObjectKey,
+	clusterKey ctrlclient.ObjectKey,
 ) error {
 	log := ctrl.LoggerFrom(ctx).WithValues(
 		"holderRef", holderRef,
 	)
-
+	cluster := &capiv1.Cluster{}
+	if err := h.client.Get(ctx, clusterKey, cluster); err != nil {
+		return err
+	}
+	defaultAPICertSANs := getDefaultAPIServerSANs(cluster)
```

Comment:
IIRC from the call last week, we wanted to also set defaults in the API either [here](https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/blob/7092a4c3ebfbe08d2f50a9a87209f4b5c14ea186/pkg/handlers/generic/clusterconfig/variables.go#L43) (if we get the required info of what infra provider it is) or [here](https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/blob/7092a4c3ebfbe08d2f50a9a87209f4b5c14ea186/api/v1alpha1/clusterconfig_types.go#L50), that way it shows up if the user doesn't set anything.
We also talked about adding the defaults in the handler as you have here so that the user doesn't need also worry about the defaults and break the cluster.

Reply:
That's not what I gathered. I thought this would be set no matter what because the wrong settings or not including these could break the cluster, which would need to be done in the handler.

Reply:
I was thinking about this some more, I wonder if we are working around the limitation of not having a mutating webhook.

What we want is to show in the API what values are being set.
Setting it in openapi schema works for defaults, but then when a user overrides it we get these implicit defaults getting added in the handler then then doesn't show up in the `Cluster` object.

Should we finally work on adding the webhook? It would help with this work and other similar scenarios.

Reply:
Yes! this should definitely be a webhook, but I think that requires a bigger scope than this.

Reply:
> Setting it in openapi schema works for defaults, but then when a user overrides it we get these implicit defaults getting added in the handler then then doesn't show up in the Cluster object

Can you please explain a bit further? what needed to show up in `Cluster` object ? Wont extra SANs will be updated in `KubeadmControlPlaneTemplate` when the mutator adds them?

Reply:
Thanks for talking with me and explaining how webhook could work.